airflow/contrib/operators/adls_to_gcs.py

```diff
@@ -0,0 +1,144 @@
+
+         The following Operator would copy all parquet files from ADLS
+        path ``/hello/world``to the GCS bucket ``mybucket``. ::
+            copy_world_files = AdlsToGoogleCloudStorageOperator(
+                task_id='copy_world_files',
+                path='hello/world/*.parquet',
+                dest_gcs='gs://mybucket',
+                replace=False,
+                azure_data_lake_conn_id='azure_data_lake_default',
+                google_cloud_storage_conn_id='google_cloud_default'
+            )
+    """
+    template_fields = ('path', 'dest_gcs')
+    ui_color = '#f0eee4'
+
+    @apply_defaults
+    def __init__(self,
+                 path,
+                 dest_gcs=None,
+                 replace=False,
+                 azure_data_lake_conn_id='azure_data_lake_default',
+                 google_cloud_storage_conn_id='google_cloud_default',
+                 delegate_to=None,
+                 *args,
+                 **kwargs):
+
+        super(AdlsToGoogleCloudStorageOperator, self).__init__(
+            path=path,
+            azure_data_lake_conn_id=azure_data_lake_conn_id,
+            *args,
+            **kwargs
+        )
+        self.dest_gcs = dest_gcs
+        self.replace = replace
+        self.google_cloud_storage_conn_id = google_cloud_storage_conn_id
+        self.delegate_to = delegate_to
+
+    def execute(self, context):
+        # use the super to list all files in an Azure Data Lake path
+        files = super(AdlsToGoogleCloudStorageOperator, self).execute(context)
+        g_hook = GoogleCloudStorageHook(
+            google_cloud_storage_conn_id=self.google_cloud_storage_conn_id,
+            delegate_to=self.delegate_to)
+
+        if not self.replace:
+            # if we are not replacing -> list all files in the ADLS path
+            # and only keep those files which are present in
+            # ADLS and not in Google Cloud Storage
+            bucket_name, prefix = _parse_gcs_url(self.dest_gcs)
+            existing_files = g_hook.list(bucket=bucket_name, prefix=prefix)
+            files = set(files) - set(existing_files)
+
+        if files:
+            hook = AzureDataLakeHook(
+                azure_data_lake_conn_id=self.azure_data_lake_conn_id
+            )
+
+            for file in files:
+                with NamedTemporaryFile(mode='wb', delete=True) as f:
+                    hook.download_file(local_path=f.name, remote_path=file)
+                    f.flush()
+                    dest_gcs_bucket, dest_gcs_prefix = _parse_gcs_url(self.dest_gcs)
+                    dest_path = os.path.join(dest_gcs_prefix, file)
+                    self.log.info("Saving file to %s", dest_path)
+
+                    g_hook.upload(bucket=dest_gcs_bucket, object=dest_path, filename=f.name)
+
+            self.log.info("All done, uploaded %d files to GCS", len(files))
+        else:
+            self.log.info("In sync, no files needed to be uploaded to GCS")
+
+        return files
```

Comment:
Wouldn't this be too big for XCOM?

Reply:
Since it's just a collection of strings, it would be pretty unlikely. XCOM values are stored as sqlalchemy.LargeBinary, correct?